"@nrwl/storybook:storybook" uses proxy package for "@storybook/core-server"

## Current Behavior
The `storybook` executor [imports from `@storybook/core`](https://github.com/nrwl/nx/blob/a32d46c5a36246b2349d72f7b50f7f0e532744bb/packages/storybook/src/executors/storybook/storybook.impl.ts#L2) to spin up a dev server. As this package is used as a proxy for backwards compatibility, I don't think it is good practice. `@storybook/core` is only installed when using `@storybook/addon-essentials` (`@storybook/addon-docs` to me more specific) and therefor breaks if storybook is used without the docs addon.

## Expected Behavior
`storybook` executor should use `@storybook/core-server` instead of relying on a proxy package.

## Steps to Reproduce
- Uninstall `@storybook/addon-essentials` from a workspace with a set up storybook project
- run `nx storybook [project]`

### Failure Logs
`Cannot find module '@storybook/core/server`

### Environment
   Node : 16.13.0
   OS   : darwin arm64
   yarn : 1.22.17
   
   nx : 13.8.7
   @nrwl/angular : undefined
   @nrwl/cli : 13.8.7
   @nrwl/cypress : 13.8.7
   @nrwl/detox : undefined
   @nrwl/devkit : 13.8.7
   @nrwl/eslint-plugin-nx : 13.8.7
   @nrwl/express : undefined
   @nrwl/jest : 13.8.7
   @nrwl/js : 13.8.7
   @nrwl/linter : 13.8.7
   @nrwl/nest : undefined
   @nrwl/next : undefined
   @nrwl/node : undefined
   @nrwl/nx-cloud : undefined
   @nrwl/react : 13.8.7
   @nrwl/react-native : undefined
   @nrwl/schematics : undefined
   @nrwl/storybook : 13.8.7
   @nrwl/tao : 13.8.7
   @nrwl/web : 13.8.7
   @nrwl/workspace : 13.8.7
   typescript : 4.5.5
   rxjs : 6.6.7

Hi there @michaelcamper ! I am just trying out your issue now! So, your issue exists, indeed if I remove `@storybook/addon-essentials` I get the `Cannot find module '@storybook/core/server'` error.

However, if I change the import to `@storybook/core-server`, I still get the not found error, for `core-server` this time. Do you want to try it out on your own, too, and let me know if it works for you in a different way? I can help you out make the change to the Nx codebase and run it locally to test if you want.

--- a/package.json
+++ b/package.json
@@ -71,7 +71,6 @@
     "@nrwl/react": "13.10.0-beta.6",
     "@nrwl/web": "13.10.0-beta.6",
     "@nrwl/workspace": "13.10.0-beta.6",
-    "nx": "13.10.0-beta.6",
     "@parcel/watcher": "2.0.4",
     "@phenomnomnominal/tsquery": "4.1.1",
     "@pmmmwh/react-refresh-webpack-plugin": "^0.5.1",
@@ -87,10 +86,11 @@
     "@storybook/addon-knobs": "~6.3.0",
     "@storybook/angular": "~6.4.12",
     "@storybook/core": "~6.4.12",
+    "@storybook/core-server": "^6.4.19",
     "@storybook/react": "~6.4.12",
     "@svgr/webpack": "^6.1.2",
-    "@swc/core": "^1.2.152",
     "@swc-node/register": "^1.4.2",
+    "@swc/core": "^1.2.152",
     "@testing-library/react": "11.2.6",
     "@testing-library/react-hooks": "7.0.1",
     "@tippyjs/react": "^4.2.6",
@@ -192,6 +192,7 @@
     "ng-packagr": "~13.3.0",
     "ngrx-store-freeze": "0.2.4",
     "node-fetch": "^2.6.7",
+    "nx": "13.10.0-beta.6",
     "open": "^8.4.0",
     "parse-markdown-links": "^1.0.4",
     "parse5": "4.0.0",
@@ -309,4 +310,3 @@
     "immer": "~9.0.6"
   }
 }
-

--- a/packages/storybook/migrations.json
+++ b/packages/storybook/migrations.json
@@ -81,6 +81,15 @@
     }
   },
   "packageJsonUpdates": {
+    "13.10.0": {
+      "version": "13.10.0-beta.0",
+      "packages": {
+        "@storybook/core-server": {
+          "version": "~6.4.12",
+          "alwaysAddToPackageJson": true
+        }
+      }
+    },
     "13.4.5": {
       "version": "13.4.5-beta.1",
       "packages": {

--- a/packages/storybook/src/executors/storybook/storybook.impl.spec.ts
+++ b/packages/storybook/src/executors/storybook/storybook.impl.spec.ts
@@ -3,10 +3,10 @@ import * as fs from 'fs';
 
 import { ExecutorContext } from '@nrwl/devkit';
 
-jest.mock('@storybook/core/server', () => ({
+jest.mock('@storybook/core-server', () => ({
   buildDevStandalone: jest.fn().mockImplementation(() => Promise.resolve()),
 }));
-import { buildDevStandalone } from '@storybook/core/server';
+import { buildDevStandalone } from '@storybook/core-server';
 
 import storybookExecutor, { StorybookExecutorOptions } from './storybook.impl';
 import { join } from 'path';

--- a/packages/storybook/src/executors/storybook/storybook.impl.ts
+++ b/packages/storybook/src/executors/storybook/storybook.impl.ts
@@ -1,5 +1,5 @@
 import { ExecutorContext, logger } from '@nrwl/devkit';
-import { buildDevStandalone } from '@storybook/core/server';
+import { buildDevStandalone } from '@storybook/core-server';
 import 'dotenv/config';
 import { showStorybookV5Warning } from '../../utils/utilities';
 import { CommonNxStorybookConfig } from '../models';

--- a/packages/storybook/src/generators/init/init.spec.ts
+++ b/packages/storybook/src/generators/init/init.spec.ts
@@ -40,6 +40,9 @@ describe('@nrwl/storybook:init', () => {
       expect(
         packageJson.devDependencies['@storybook/addon-essentials']
       ).toBeDefined();
+      expect(
+        packageJson.devDependencies['@storybook/core-server']
+      ).toBeDefined();
 
       // angular specific
       expect(packageJson.devDependencies['@storybook/angular']).toBeDefined();

--- a/packages/storybook/src/generators/init/init.ts
+++ b/packages/storybook/src/generators/init/init.ts
@@ -38,6 +38,13 @@ function checkDependenciesInstalled(host: Tree, schema: Schema) {
    */
 
   if (
+    !packageJson.dependencies['@storybook/core-server'] &&
+    !packageJson.devDependencies['@storybook/core-server']
+  ) {
+    devDependencies['@storybook/core-server'] = storybookVersion;
+  }
+
+  if (
     !packageJson.dependencies['@storybook/addon-essentials'] &&
     !packageJson.devDependencies['@storybook/addon-essentials']
   ) {

--- a/scripts/depcheck/missing.ts
+++ b/scripts/depcheck/missing.ts
@@ -87,6 +87,7 @@ const IGNORE_MATCHES = {
     '@storybook/addon-knobs',
     '@storybook/addon-essentials',
     '@storybook/core',
+    '@storybook/core-server',
     'rxjs',
   ],
   nx: [

